update doc for MPP utilization optimization (#7372)
pingcap/docs: tiflash/tiflash-configuration.md

    cop_pool_size = 0
    ## New in v5.0. This item specifies the maximum number of batch requests that TiFlash Coprocessor executes at the same time. If the number of requests exceeds the specified value, the exceeded requests will queue. If the configuration value is set to 0 or not set, the default value is used, which is twice the number of physical cores.
    batch_cop_pool_size = 0

>     ## New in v5.4.0. This item enables or disables the elastic thread pool function that significantly improves CPU utilization in high concurrency scenarios of TiFlash. The default value is false. The elastic thread pool function is experimental and cannot be enabled in production environments.

Reviewer: ```suggestion
    ## New in v5.4.0. This item enables or disables the elastic thread pool feature, which significantly improves CPU utilization in high concurrency scenarios of TiFlash. The default value is false. The elastic thread pool feature is experimental and not recommended to enable it in production environments.
```

Reviewer: 1. "that" is usually used for restrictive clauses. See https://www.myenglishteacher.eu/blog/which-vs-that/
2. In TiDB docs, "feature" is usually when it means "function" to avoid confusion with SQL functions.

>     ## New in v5.4.0. This item enables or disables the elastic thread pool feature, which significantly improves CPU utilization in high concurrency scenarios of TiFlash. The default value is false. The elastic thread pool feature is experimental and is not recommended for production environments.

Reviewer: ```suggestion
    ## New in v5.4.0. This item enables or disables the elastic thread pool feature, which significantly improves CPU utilization in high concurrency scenarios of TiFlash. The default value is false. The elastic thread pool feature is experimental and not recommended for production environments.
```
